model_managers/model_manager.py

```diff
@@ -488,6 +702,170 @@ def get_class_ct(self, class_: str, ct_prop_name='rdfs:label'):
         else:
             return []
 
+    def get_class_ct_map(self, classes: list, ct_props: list = None, identifier='label'):
+        """
+        Return controlled terminology for a given list of classes.
+        :param classes: list of class labels of class.identifier property values if using custom identifier
+                        for example: with identifier = 'short_label', classes = ['shortlabel1', 'shortlabel2']
+        :param ct_props: list of controlled terminology props to collect eg:
+                         ['label', 'Codelist Code', 'Order']
+                         defaults to ['rdfs:label']
+        :param identifier: string used when identifying classes
+        :return: Dictionary of classes and found controlled terminology for example:
+                 with classes = ['class1', 'class2'] and ct_props = ['label'] a typical result might be:
+                {'class1': [{'label': 'term1'}, {'label': 'term2'}], 'class2': [{{'label': 'term3'}}]}
+                Where 'class1' has two terms labeled 'term1' and 'term2' and 'class2' has one term labeled 'term3'
+        """
+
+        if ct_props is None:
+            ct_props = ['rdfs:label']
+        elif type(ct_props) == str:
+            ct_props = [ct_props]
+
+        prop_collection = f'["{ct_props[0]}", term.`{ct_props[0]}`]'
+        if len(ct_props) != 1:
+            for prop in ct_props[1:]:
+                prop_collection += f', ["{prop}", term.`{prop}`]'
+
+        q = f"""
+        UNWIND $classes as class
+        MATCH (c:Class)-[:HAS_CONTROLLED_TERM]->(term:Term)
+        WHERE c.`{identifier}` = class
+        WITH c, collect(apoc.map.fromPairs([{prop_collection}])) as terms
+        RETURN apoc.map.setKey({{}}, c.`{identifier}`, terms) as ct
+        """
+
+        params = {'classes': classes}
+        res = self.query(q, params)
+
+        data = {}
+        if res:
+            for term_dict in res:
+                # Flattened result {class.identifer:[[term.ct_props[0], ...], [term.ct_props[0], ...]]}
+                data.update(term_dict.get('ct'))
+        return data
+
+    def delete_ct(self, controlled_terminology: dict, term_props: list, identifier='label'):
```

Comment:
in the function above a property named term_props is named ct_props - please align on naming if these are the same things, also it would allow to avoid confusion with 
`UNWIND $terminology[class_label] as term_props` in the query